### ISSUE TITLE
Could not find pull request warning for pull_request closed event

### ISSUE THREAD

**Issue by convers39** [actions/labeler#775]
**Description:**

I need to run a workflow after a PR is closed or merged. So I set the `types` as `closed`. Now the issue is since the PR is closed first in order to trigger the workflow, labeler seems not able to find a closed PR and gives me a warning message of `Could not find pull request <number>, skipping`. I have some tasks which rely on labeler, so wonder if there is any way to work with closed PRs.

```yml
on:
  pull_request:
    types: [ closed ]
jobs:
  delete_channel:
    runs-on: ubuntu-latest
    permissions:
      contents: read
      id-token: write
    steps:
      - name: チェックアウト
        uses: actions/checkout@v4
        with:
          ref: ${{ github.event.pull_request.head.sha }}
      # ...
      - id: pr_labels
        uses: actions/labeler@v5

```

<img width="879" alt="image" src="https://github.com/actions/labeler/assets/42597395/5a4e6626-2680-4433-8144-ac2dcfdb4d5c">

**Action version:**
Specify the action version

v5

**Platform:**
- [ ] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [ ] Hosted
- [ ] Self-hosted

**Repro steps:**  

basically the action file I posted 

**Expected behavior:**

Possible to deal with closed PRs.

**Actual behavior:**

`Could not find pull request <number>, skipping` warning, and the following steps are skipped.

**Comment by convers39** [comment]
currently, I tried another workflow that can retrieve the labels after PR is closed

https://github.com/marketplace/actions/get-labels-action

**Comment by HarithaVattikuti** [comment]
Hello @convers39 
Thank you for creating this issue. We will investigate it and get back to you as soon as we have some feedback.

**Comment by abcdefghiraj** [comment]
Try adding the following to the params, I went through the code quickly and it seems prNumber is retrieved from a param `pr-number`. 
```
pr-number: ${{ github.event.number }}
```

**Comment by suyashgaonkar** [comment]
Hi @convers39 , We tried to investigate the issue by running the provided workflow and we see the error mentioned in the screenshot below. 

<img width="1495" alt="Image" src="https://github.com/user-attachments/assets/8f8fb51c-861e-441b-87c2-5adf3b3d5efc" />

Following is the failed [run](https://github.com/setup-actions-demo/labeler-test/actions/runs/14377213443). The error was resolved by adding `pull-requests: write` permission in the workflow. Please refer to the following [run](https://github.com/setup-actions-demo/labeler-test/actions/runs/14377488990) that was successful after adding permissions. You can refer to following documentation [link](https://github.com/actions/labeler#recommended-permissions) for more details on the required permissions for actions/labeler.

**Comment by suyashgaonkar** [comment]
Hi @convers39 , we are waiting for your response on the following ticket.

**Comment by suyashgaonkar** [comment]
Hi @convers39 , we are waiting for your response.

**Comment by suyashgaonkar** [comment]
Hi, this issue has been inactive for some time. It seems that the issue may have been resolved or is no longer relevant. I will proceed to close this issue. Feel free to reach us out incase of any further updates. Thank you!
